src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -212,7 +212,11 @@ public static Calendar convertToUserTimeZone(Calendar time, double timeZone) {
      * Converts the {@code localDate} from {@code localTimeZone} to UTC through shifting by the offset.
      * Does not shift if {@code localDate} is a special representation.
      */
+    @Deprecated
```

Comment:
To prevent future messing up, I think you should add a warning of not to remove this function until we have migrated the database and also write a test for this behaviour.